### PR TITLE
Cae 112

### PR DESCRIPTION
**Pull request by nicklundin-bl**
#### Required
- [x] I ran the CI build script to ensure that the code passes the following
    - Unit tests
    - Integration tests
    - Linting rules
- [ ] I ran the E2E test suite locally to ensure that no regressions were introduced

#### Optional
- [ ] My PR contains a DB schema change
    - Reviewers pay special attention to queries/indices when this box is checked
- [ ] My PR is dependent on a change to our infrastructure
    - New SSM keys
    - IAM role updates
    - ETC
- [x] My pull request is fewer than 400 lines of code

## Purpose
CAE-112 adds the put endpoint for classroom demographics

### CODE REVIEW

File: src/main/java/com/betterlesson/learningwalk/errors/exceptions/LabDataSerializationException.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.betterlesson.learningwalk.errors.exceptions;
```
Comment (nicklundin-bl) [inline src/main/java/com/betterlesson/learningwalk/errors/exceptions/LabDataSerializationException.java:1]: This file was moved (everything in the /errors dir was moved to the appropriate package level)

File: src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkService.java [hunk 2]
```diff
@@ -28,6 +30,34 @@ public class LearningWalkService {
...
+  public LearningWalkClassroom saveClassroomDemographics(
+      UUID lwID, Long classroomId, SaveLearningWalkClassroomDemographicsRequest demographics) {
+    LearningWalk learningWalk =
+        learningWalkRepository
+            .findByUuid(lwID)
+            .orElseThrow(() -> new LearningWalkNotFoundException(lwID));
+    List<LearningWalkClassroom> learningWalkClassrooms =
+        learningWalkClassroomsRepository.findByLearningWalkId(learningWalk.getId());
+
+    LearningWalkClassroom classroom =
```
Comment (rishipisipati) [inline src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkService.java:43]: Why not just search by classroom id and throw an exception if not found? 
`
LearningWalkClassroom classroom = learningWalkClassroomRepository.findById(classroomId).orElseThrow(() -> new LearningWalkClassroomNotFoundException(classroomId));`

File: src/test/java/com/betterlesson/learningwalk/services/LabServiceTests.java [hunk 3]
```diff
@@ -145,8 +144,8 @@ public void should_return_lab_data_if_the_lab_data_is_good() throws Exception {
     // No point testing a third party serializer, so I am mocking out the response here
     when(mockObjectMapper.readValue(any(FileInputStream.class), eq(LabDataForLearningWalk.class)))
         .thenReturn(getMockLabDataForLearningWalk());
-    // Unit test should fail if any exceptions are being thrown at this point. 
+    // Unit test should fail if any exceptions are being thrown at this point.
     mockLabService.getLabDataForLearningWalk(123L, 123L, "");
-    assert(true);
+    assert (true);
```
Comment (rishipisipati) [inline src/test/java/com/betterlesson/learningwalk/services/LabServiceTests.java:149]: Extra space?

File: src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkService.java [hunk 4]
```diff
@@ -28,6 +30,34 @@ public class LearningWalkService {
...
+        learningWalkClassrooms.stream()
+            .filter(x -> x.getId().equals(classroomId))
+            .findFirst()
+            .orElseThrow(() -> new LearningWalkClassroomsNotFound(learningWalk.getId()));
```
Comment (rishipisipati) [inline src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkService.java:47]: LearningWalkClassroomsNotFoundException to maintain consistency.

File: src/test/java/com/betterlesson/learningwalk/services/LabServiceTests.java (hunk 3 above)
Comment (nicklundin-bl) [reply src/test/java/com/betterlesson/learningwalk/services/LabServiceTests.java:149]: I ran the formatter. I think `assert (true)` is the format for naked java `assert` while `assertTrue(true)` will not include the space. 🤷

**Review by rishipisipati — APPROVED** [review]

**Pull request merged**